crates/core/src/ffi/mod.rs

```diff
@@ -252,7 +274,8 @@ pub extern "C" fn document_merge_fragment_json<'a>(
             };
         }
     };
-    let root : Root = match other_fragment.try_into() {
+    let doc = unsafe { &mut *doc };
+    let new_root = match doc.merge_fragment(root_diff) {
```

Comment:
The order here is a bit whack and I don't love it. This call to `merge_fragment` merges the [`RootDiff`](https://github.com/liveview-native/liveview-native-core/blob/d9713724988b213447a02400dae7185ee1bfb5a2/crates/core/src/diff/fragment/mod.rs#L7-L12) into the [`Root`](https://github.com/liveview-native/liveview-native-core/blob/d9713724988b213447a02400dae7185ee1bfb5a2/crates/core/src/diff/fragment/mod.rs#L14-L20) from the `fragment` module which is not the same as [`Document.root`](https://github.com/liveview-native/liveview-native-core/blob/d9713724988b213447a02400dae7185ee1bfb5a2/crates/core/src/dom/mod.rs#L62). After this, it's then stringified and passed to [`document_merge`](https://github.com/liveview-native/liveview-native-core/blob/d9713724988b213447a02400dae7185ee1bfb5a2/crates/core/src/ffi/mod.rs#L187-L231).

Over in https://github.com/liveview-native/liveview-native-core/pull/43, I've concluded that the document merge call will require some refactoring (and breaking changes) so I'm putting off the refactor for now.